tcms/report/data.py

```diff
@@ -152,52 +149,89 @@ def case_runs_count(self, component_id):
         subtotal['TOTAL'] = total
         return subtotal
 
+    @staticmethod
+    def total_cases(product_id, condition=None):
```

Comment:
You should add docstrings for every function/method that you modify and describe the type of the expected parameter.